base/asyncevent.jl

```diff
@@ -284,7 +285,8 @@ function timedwait(testcb::Function, secs::Float64; pollint::Float64=0.1)
 
     if !testcb()
         t = Timer(timercb, pollint, interval = pollint)
-        ret = fetch(done)::Symbol
+        ret, e = fetch(done)
+        ret === :error && rethrow(e)
```

Comment:
`rethrow` doesn't work outside a catch block (it's throwing `ERROR: rethrow(exc) not allowed outside a catch block`) so you have to use `throw`.

Reply:
Bad mistake on my part. Unfortunately this means that the exception thrown won't have the original stacktrace

Reply:
Only if the callback raises an exception on the first call will generate the original stack trace. Exceptions raised after will not have the original stack trace. This seems better than silencing all exceptions from the callback.

Reply:
Alternatively I can have `timedwait` return `:error` again and just capture exceptions from the first `testcb` call

Reply:
You could capture the stack trace and store it in a CapturedException?

Reply:
I've updated the code to always return a `CapturedException`